exercises/poker/src/test/java/PokerTest.java

```diff
@@ -6,158 +6,234 @@
+		assertEquals(Arrays.asList(hand), new Poker(Arrays.asList(hand)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void highestCardWins() {
+		String highest8 = "4D 5S 6S 8D 3C";
+		String highest10 = "2S 4C 7S 9H 10H";
+		String highestJ = "3S 4S 5D 6H JH";
+		assertEquals(Arrays.asList(highestJ), new Poker(Arrays.asList(highest8, highest10, highestJ)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void tieHasMultipleWinners() {
+		String highest8 = "4D 5S 6S 8D 3C";
+		String highest10 = "2S 4C 7S 9H 10H";
+		String highestJh = "3S 4S 5D 6H JH";
+		String highestJd = "3H 4H 5C 6C JD";
+		assertEquals(Arrays.asList(highestJh, highestJd),
+				new Poker(Arrays.asList(highest8, highest10, highestJh, highestJd)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void sameHighCards() {
+		String nextHighest3 = "3S 5H 6S 8D 7H";
+		String nextHighest2 = "2S 5D 6D 8C 7S";
+		assertEquals(Arrays.asList(nextHighest3), new Poker(Arrays.asList(nextHighest3, nextHighest2)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void nothingVsOnePair() {
+		String nothing = "4S 5H 6C 8D KH";
+		String pairOf4 = "2S 4H 6S 4D JH";
+		assertEquals(Arrays.asList(pairOf4), new Poker(Arrays.asList(nothing, pairOf4)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void twoPairs() {
+		String pairOf2 = "4S 2H 6S 2D JH";
+		String pairOf4 = "2S 4H 6C 4D JD";
+		assertEquals(Arrays.asList(pairOf4), new Poker(Arrays.asList(pairOf2, pairOf4)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void onePairVsDoublePair() {
+		String pairOf8 = "2S 8H 6S 8D JH";
+		String doublePair = "4S 5H 4C 8C 5H";
+		assertEquals(Arrays.asList(doublePair), new Poker(Arrays.asList(pairOf8, doublePair)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void twoDoublePairs() {
+		String doublePair2And8 = "2S 8H 2D 8D 3H";
+		String doublePair4And5 = "4S 5H 4C 8S 5D";
+		assertEquals(Arrays.asList(doublePair2And8),
+				new Poker(Arrays.asList(doublePair2And8, doublePair4And5)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void sameHighestPair() {
+		String doublePair2AndQ = "2S QS 2C QD JH";
+		String doublePairJAndQ = "JD QH JS 8D QC";
+		assertEquals(Arrays.asList(doublePairJAndQ),
+				new Poker(Arrays.asList(doublePairJAndQ, doublePair2AndQ)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void identicallyRankedPairs() {
+		String kicker8 = "JD QH JS 8D QC";
+		String kicker2 = "JS QS JC 2D QD";
+		assertEquals(Arrays.asList(kicker8), new Poker(Arrays.asList(kicker8, kicker2)).getBestHands());
+	}
+
+	@Ignore("Remove to run test")
+	@Test
+	public void doublePairVsThree() {
+		String doublePair2And8 = "2S 8H 2H 8D JH";
+		String threeOf4 = "4S 5H 4C 8S 4H";
+		assertEquals(Arrays.asList(threeOf4), new Poker(Arrays.asList(doublePair2And8, threeOf4)).getBestHands());
```

Comment:
When there's only one object in the list you can use `Collections.singletonList` instead of `Arrays.asList` 🙂